src/mame/misc/silverball.cpp

```diff
@@ -41,35 +49,33 @@ class silverball_state : public driver_device
 
 private:
 	void mem_map(address_map &map);
-	void io_map(address_map &map);
 
 	required_device<cpu_device> m_maincpu;
 };
 
 void silverball_state::mem_map(address_map &map)
 {
 	map(0x00000000, 0x0009ffff).ram();
-	map(0xfffc0000, 0xffffffff).rom().region("mb_bios", 0);
-}
-
-void silverball_state::io_map(address_map &map)
-{
+	map(0x000e0000, 0x000fffff).rom().region("bios", 0x20000);
+	map(0xfffc0000, 0xffffffff).rom().region("bios", 0);
 }
 
 static INPUT_PORTS_START(silverball)
 INPUT_PORTS_END
 
 void silverball_state::silverball(machine_config &config)
 {
-	PENTIUM(config, m_maincpu, 133'000'000);
+	PENTIUM(config, m_maincpu, 133'000'000); // Pentium-S minimum, up to a Pentium II
 	m_maincpu->set_addrmap(AS_PROGRAM, &silverball_state::mem_map);
-	m_maincpu->set_addrmap(AS_IO, &silverball_state::io_map);
+
+	PCI_ROOT(config, "pci", 0);
+	// ...
 }
 
 
 // Silverball machines used different motherboards. By now, we're adding all the known BIOSes here
```

Comment:
The trouble here is that handling all these different motherboards with different CPU sockets and architectures cannot possibly work in how MAME handles things down the road. We can think about slotify motherboards but given we don't even have an option to slotify CPUs I guess shooting a tracer with romset splits is the best option for now.